python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -181,6 +222,14 @@ def end_training():
 
             return output
 
+        futures = self.worker_group.execute_async(pause_reporting)
+        self.get_with_failure_handling(futures)
+
+        # Finish up processing results.
+        results = self.fetch_next_result()
+        while results is not None:
```

Comment:
If we disable reporting will `results` ever not be `None`?

Reply:
Yep there's a corner case if there is a reported result in the queue immediately before `pause_reporting` is called. I'll update the doc for `pause_reporting` to mention this.